File: app/src/main/java/fr/legrand/oss117soundboard/presentation/ui/activity/BaseActivity.java
package fr.legrand.oss117soundboard.presentation.ui.activity;

import android.app.Activity;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.view.View;

import fr.legrand.oss117soundboard.presentation.OSSApplication;
import fr.legrand.oss117soundboard.presentation.di.component.ActivityComponent;
import fr.legrand.oss117soundboard.presentation.di.component.DaggerActivityComponent;
import fr.legrand.oss117soundboard.presentation.di.module.ActivityModule;

/**
 * Created by Benjamin on 30/09/2017.
 */

public abstract class BaseActivity extends Activity {

    private ActivityComponent activityComponent;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initializeInjector();
    }

    private void initializeInjector() {
        activityComponent = DaggerActivityComponent.builder().applicationComponent(((OSSApplication) getApplication()).getApplicationComponent())
                .activityModule(new ActivityModule(this)).build();
    }

    public ActivityComponent getActivityComponent() {
        return activityComponent;
    }

    public abstract View getRootView();

}

File: app/src/main/java/fr/legrand/oss117soundboard/presentation/ui/base/BaseActivity.kt
package fr.legrand.oss117soundboard.presentation.ui.base

import android.os.Bundle
import android.support.annotation.IdRes
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.View
import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject

/**
 * Created by Benjamin on 30/09/2017.
 */

abstract class BaseActivity : AppCompatActivity(), HasSupportFragmentInjector {

    @Inject
    lateinit var fragmentInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
    }

    @LayoutRes
    abstract fun getLayoutId(): Int

    @IdRes
    abstract fun getRootView() : View

    override fun supportFragmentInjector(): AndroidInjector<Fragment> = fragmentInjector

}
